shell.c

```diff
@@ -306,16 +309,94 @@ char* readLine() {
     }
 }
 
+// Display result of the assignment
+struct pre_assessment {
+    bool valid;
+    char *assignment;
+};
+
+void display_assignment_result(struct pre_assessment pa, char *line) {
+    // Retrieve the first 8 characters forming the "echo -e "
+    const int COMM_SIZE = 8;
+    char command[COMM_SIZE + 1];
+    for (int i = 0; i < COMM_SIZE; i++) command[i] = line[i];
```

Comment:
strndup could be used.

Reply:
Thanks, I will definitely look more onto that.